Client/Frontend/Library/Bookmarks/BookmarksPanelViewModel.swift

```diff
@@ -61,6 +61,26 @@ class BookmarksPanelViewModel {
         flashLastRowOnNextReload = true
     }
 
+    func moveRowAt(moveRowAt sourceIndexPath: IndexPath, to destinationIndexPath: IndexPath) {
+        guard let bookmarkNode = bookmarkNodes[safe: sourceIndexPath.row] else {
+            return
+        }
+
+        let newIndex = getNewIndex(from: destinationIndexPath.row)
+        _ = profile.places.updateBookmarkNode(guid: bookmarkNode.guid, position: UInt32(newIndex))
+
+        bookmarkNodes.remove(at: sourceIndexPath.row)
+        bookmarkNodes.insert(bookmarkNode, at: destinationIndexPath.row)
+    }
+
+    // MARK: - Private
+
+    /// Since we have a Desktop folder that isn't referenced in A-S under the mobile folder, we need to account for this when updating bookmark index.
+    /// Index is also inverted inside the mobile folder.
+    private func getNewIndex(from index: Int) -> Int {
+        return bookmarkFolderGUID == BookmarkRoots.MobileFolderGUID ? bookmarkNodes.count - index - 1 : index
```

Comment:
This could be a good candidate to unit test.
Can you add documentation to make `bookmarkNodes.count - index - 1` easier to understand